pkg/kubectl/cmd/util/factory.go

```diff
@@ -254,8 +254,8 @@ func NewFactory(optionalClientConfig clientcmd.ClientConfig) *Factory {
 			}
 			return kubectl.ReaperFor(mapping.Kind, client, expClient)
 		},
-		Validator: func() (validation.Schema, error) {
-			if flags.Lookup("validate").Value.String() == "true" {
+		Validator: func(validate bool) (validation.Schema, error) {
```

Comment:
Correction: the caller gets `schema` back from `Validator()` and sets the schema of resource builder. Therefore, it makes the code cleaner to always get a schema back, and inside `Validator()` we use `shouldValidate` to determine which schema (swagger or null) to be returned.